src/psyclone/domain/nemo/transformations/nemo_loop_fuse.py

```diff
@@ -108,12 +109,20 @@ def validate(self, node1, node2, options=None):
             if var_info1.is_read_only() and var_info2.is_read_only():
                 continue
 
-            if var_name in symbol_table:
+            try:
                 # Find the symbol for this variable. We only need to check
                 # one symboltable.
                 symbol = symbol_table.lookup(var_name)
-                is_array = symbol.is_array
-            else:
+                if isinstance(symbol, DataSymbol):
+                    is_array = symbol.is_array
+                else:
+                    # This typically indicates a symbol is used that we do
+                    # not have detailled information for, e.g. based on a
+                    # generic 'use some_mod' statement. In thise case use
+                    # the information based on the access pattern, which
+                    # is at least better than having no information at all.
```

Comment:
As discussed on Teams, I think this is good enough for this PR but please could you add a note that we could chase down the definition of the symbol. In contrast to what I said in Teams, we don't quite have *all* the necessary functionality for this since we would need to get a list of *all* Container symbols with 'wildcard' imports and then search each of them in turn, probably recursively. I expect we will eventually need this functionality in other places too so please could you create an Issue for it.

Reply:
Done, #1213.